tests/component-parser.spec.ts

```diff
@@ -0,0 +1,109 @@
+  expect(serialize(parse(`BookItem[name |= 'foo']`))).toBe(`BookItem[name |= "foo"]`);
+});
+
+it('should tolerate spacing', async () => {
+  expect(serialize(parse(` BookItem[ name = "Foo "  ]`))).toBe('BookItem[name = "Foo "]');
+  expect(serialize(parse(`  BookItem  [  name = "Foo"  ]   `))).toBe('BookItem[name = "Foo"]');
+  expect(serialize(parse(`  [    name = "Foo"]`))).toBe('[name = "Foo"]');
+  expect(serialize(parse(`  BookItem  ["name" = "Foo" i]   `))).toBe('BookItem[name = "Foo" i]');
+  expect(serialize(parse(`BookItem  [   'name' = 'Foo'i ]   `))).toBe('BookItem[name = "Foo" i]');
+});
+
+it('should escape', async () => {
+  expect(serialize(parse(`BookItem['jake\\'s' = 10]`))).toBe(`BookItem["jake's" = 10]`);
+  expect(serialize(parse(`BookItem['jake"s' = 10]`))).toBe(`BookItem["jake\\"s" = 10]`);
+  expect(serialize(parse(`BookItem["jake\\"s" = 10]`))).toBe(`BookItem["jake\\"s" = 10]`);
+  expect(serialize(parse(`BookItem[name = 'foo\\'bar']`))).toBe(`BookItem[name = "foo'bar"]`);
+  expect(serialize(parse(`BookItem[name = "foo'bar"]`))).toBe(`BookItem[name = "foo'bar"]`);
+  expect(serialize(parse(`BookItem[name = "foo\\"bar"]`))).toBe(`BookItem[name = "foo\\"bar"]`);
+});
+
+it('should parse int values', async () => {
+  expect(serialize(parse(`ColorButton[value = 10]`))).toBe('ColorButton[value = 10]');
+  expect(serialize(parse(`ColorButton[value = +10]`))).toBe('ColorButton[value = 10]');
+  expect(serialize(parse(`ColorButton[value = -10]`))).toBe('ColorButton[value = -10]');
+  expect(serialize(parse(`ColorButton  [ "nested". "index" = 0 ]   `))).toBe('ColorButton[nested.index = 0]');
+});
+
+it('should parse float values', async () => {
+  expect(serialize(parse(`ColorButton[value = -12.3]`))).toBe('ColorButton[value = -12.3]');
+  expect(serialize(parse(`ColorButton  ['nested'.value = 4.1]`))).toBe('ColorButton[nested.value = 4.1]');
+  expect(serialize(parse(`ColorButton  [    'nested' .value =4.1]`))).toBe('ColorButton[nested.value = 4.1]');
+});
+
+it('shoulud parse bool', async () => {
+  expect(serialize(parse(`ColorButton[enabled= false] `))).toBe('ColorButton[enabled = false]');
+  expect(serialize(parse(`ColorButton[enabled  =true] `))).toBe('ColorButton[enabled = true]');
+  expect(serialize(parse(`ColorButton[enabled  =true][ color = "red"]`))).toBe('ColorButton[enabled = true][color = "red"]');
+  expect(serialize(parse(`ColorButton[ enabled  =true][ color = "red"i][nested.index =  6]`))).toBe('ColorButton[enabled = true][color = "red" i][nested.index = 6]');
+});
+
+it('should throw on malformed selector', async () => {
+  expectError('foo[');
+  expectError('react=foo[');
+  expectError('react=foo["asd');
+  expectError('react=foo["asd"');
+  expectError('react=foo["asd"');
+
+  expectError('foo[.bar=10]');
+  expectError('foo[bar **= 10]');
+  expectError('foo[bar == 10]');
+  expectError('foo[bar = 10 [baz=20]');
+  expectError('foo[bar = 10 i[baz=20]');
+
```

Comment:
Add tests for `[foo=10 s]`, `[foo="10"s]`,  `[foo="10" s]`, `[foo="true"]`, `[foo=10 p]`, `foo.bar`, `foo[]`, `[foo="]"]`, `[foo="\\"]`, `[foo=""]`, `["a\"b"=foo]`, `[foo=""]`, `[foo=10"bar"]`, `[foo= ==]`, `[foo===]`, `[foo="=="]`, `[foo="\"]"[]`, `[foo=abc S]`, `[foo=abc \s]`, `[foo=abc"\s"]`

Reply:
Done